### ISSUE TITLE
`pdm add` creates unnecessary empty `tool.pdm` sections in pyproject.toml

### ISSUE THREAD

**Issue by nyuszika7h** [pdm-project/pdm#1464]
- [x] I have searched the issue tracker and believe that this is not a duplicate.

**Make sure you run commands with `-v` flag before pasting the output.**

## Steps to reproduce

1. `pdm init -n`
2. `pdm add minimal` (or `pdm add --group dev minimal`)

## Actual behavior

Extra empty sections are created in pyproject.toml:
```diff
diff --git a/pyproject.toml b/pyproject.toml
index 0b3175d..f72ba06 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -5,10 +5,15 @@ description = ""
 authors = [
     ...
 ]
-dependencies = []
+dependencies = [
+    "minimal>=0.1.0",
+]
 requires-python = ">=3.11"
 license = {text = "MIT"}
 
 [build-system]
 requires = ["pdm-pep517>=1.0.0"]
 build-backend = "pdm.pep517.api"
+
+[tool]
+[tool.pdm]
```

If `--group dev` is used:
```diff
diff --git a/pyproject.toml b/pyproject.toml
index 0b3175d..113a112 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -8,7 +8,15 @@ authors = [
 dependencies = []
 requires-python = ">=3.11"
 license = {text = "MIT"}
+[project.optional-dependencies]
+dev = [
+    "minimal>=0.1.0",
+]
 
 [build-system]
 requires = ["pdm-pep517>=1.0.0"]
 build-backend = "pdm.pep517.api"
+
+[tool]
+[tool.pdm]
+[tool.pdm.dev-dependencies]
```

## Expected behavior

Those sections should not be written.

## Environment Information

```bash
PDM version:
  2.1.5
Python Interpreter:
  /home/nyuszika7h/test/.venv/bin/python (3.11)
Project Root:
  /home/nyuszika7h/test
Project Packages:
  None
{
  "implementation_name": "cpython",
  "implementation_version": "3.11.0",
  "os_name": "posix",
  "platform_machine": "x86_64",
  "platform_release": "5.10.0-18-amd64",
  "platform_system": "Linux",
  "platform_version": "#1 SMP Debian 5.10.140-1 (2022-09-02)",
  "python_full_version": "3.11.0",
  "platform_python_implementation": "CPython",
  "python_version": "3.11",
  "sys_platform": "linux"
}
```

**Comment by pawamoy** [comment]
I've noticed this as well.

**Comment by frostming** [comment]
Fixed in [tomlkit 0.11.6](https://github.com/sdispater/tomlkit/releases/tag/0.11.6)
